api/src/components/affiliations/controller.ts

```diff
@@ -74,3 +73,36 @@ export const destroy = async (
         return response.json(500, { message: 'Unknown server error.' });
     }
 };
+
+export const destroyAll = async (
+    event: I.AuthenticatedAPIRequest<undefined, undefined, I.DeleteAffiliationPathParams>
+): Promise<I.JSONResponse> => {
+    try {
+        const publication = await publicationService.get(event.pathParameters.id);
+
+        //check that the publication exists
+        if (!publication) {
+            return response.json(404, {
+                message: 'This publication does not exist.'
+            });
+        }
+
+        if (publication.currentStatus === 'LIVE') {
```

Comment:
The same as above. This is allowing corresponding user to  remove affiliations while the publication is LOCKED...